### PR TITLE
Remove borrowdirect from requests and orangelight

### LINKED ISSUE #3421: Remove FlipFlop from Reshare
- We switched to the ReShare platform on December 13th. 

- Old borrow direct references need to be removed.

- ReShare should not be behind FlipFlop anymore. 
- This is a cleanup ticket

### PR DESCRIPTION
**Pull request by sandbergja**
closes #3421

### CODE REVIEW

File: spec/controllers/account_controller_spec.rb [hunk 1]
```diff
@@ -46,15 +46,15 @@
     end
 
     it 'Returns Non-canceled Illiad Transactions' do
-      get :borrow_direct_redirect
+      get :digitization_requests
```
Comment (sandbergja) [inline spec/controllers/account_controller_spec.rb:49]: These tests are for the `#illiad_patron_client` method, which is called by both `#borrow_direct_redirect` and `digitization_requests`.

File: spec/features/requests/request_spec.rb [hunk 2]
```diff
@@ -230,7 +230,6 @@
           expect(page).to have_content 'Pick-up location: Firestone Library'
           expect { click_button 'Request Selected Items' }.to change { ActionMailer::Base.deliveries.count }.by(1)
           expect(page).to have_content 'Your request was submitted. Our library staff will review the request and contact you with any questions or updates.'
-          expect(page).not_to have_content 'Request submitted to BorrowDirect'
```
Comment (sandbergja) [inline spec/features/requests/request_spec.rb:233]: I thought that this expectation was no longer very meaningful, if there is no longer a way for the application to return this string.

File: spec/features/requests/request_spec.rb [hunk 3]
```diff
@@ -1400,9 +1287,11 @@
...
         describe 'Request a temp holding item from Resource Sharing - RES_SHARE$IN_RS_REQ' do
           before do
-            borrow_direct = ::BorrowDirect::RequestItem.new("22101008199999")
-            expect(::BorrowDirect::RequestItem).to receive(:new).with("22101008199999").and_return(borrow_direct)
-            expect(borrow_direct).to receive(:make_request).with("Firestone Library", isbn: '9781472523242').and_return('123456')
+            stub_illiad_patron
+            stub_request(:post, transaction_url)
+              .to_return(status: 200, body: responses[:transaction_created], headers: {})
+            stub_request(:post, transaction_note_url)
+              .to_return(status: 200, body: responses[:note_created], headers: {})
```
Comment (sandbergja) [inline spec/features/requests/request_spec.rb:1294]: These requests are handled by ILLiad now, so let's mock ILLiad in our test setup.

**Review by kevinreiss — APPROVED** [review]

File: spec/features/requests/request_spec.rb (hunk 2 above)
Comment (kevinreiss) [reply spec/features/requests/request_spec.rb:233]: agreed

**Pull request merged**